### PR TITLE
Added fix for the infinite page refresh in resource Deployment tools page

### LINKED ISSUE #13812: infinite reloading loop when trying to load any resource deployment option
latest Insider

steps:
1. open resource deployment dialog
2. pick any option and click 'Select' button

Expected:
options and prerequisite page loads.

Actual:
the options and prerequisite page get into a infinite loading loop.

### PR DESCRIPTION
**Pull request by aasimkhan30**
Generating events for select boxes only when the select box value is changed.

This PR fixes #13812 

For some reason, I was noticing a trailing value in the events generated. 

By trailing value, I mean, the value that was last selected and not the current one.
For example:
if A is the initial value and the user changes it to B, the event returned A instead of B

### CODE REVIEW

File: extensions/resource-deployment/src/ui/toolsAndEulaSettingsPage.ts [hunk 1]
```diff
@@ -185,19 +185,22 @@ export class ToolsAndEulaPage extends ResourceTypePage {
...
 						}).component();
 
-						resourceTypeOptions.push(option.values[0]);
+						resourceTypeOptions.push(optionSelectedValue);
 
 						this.wizard.registerDisposable(optionSelectBox.onValueChanged(async () => {
-							resourceTypeOptions[index] = <ResourceTypeOptionValue>optionSelectBox.value;
-							this.wizard.provider = this.getCurrentProvider();
-							await this.wizard.open();
+							if (resourceTypeOptions[index].name !== (<ResourceTypeOptionValue>optionSelectBox.value || true).name) {
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/toolsAndEulaSettingsPage.ts:199]: Do you need this check still if the event isn't firing when it's the same option?

File: src/sql/base/browser/ui/selectBox/selectBox.ts [hunk 2]
```diff
@@ -197,6 +196,9 @@ export class SelectBox extends vsSelectBox {
 	}
 
 	public select(index: number): void {
+		if (this._selectedOption && this._optionsDictionary.get(this._selectedOption) === index) { // Not generating an event if the same value is selected.
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/selectBox/selectBox.ts:199]: I'd specifically do `this._selectedOption === undefined` - an empty value is still a valid option.

**Review by Charles-Gagnon — COMMENTED** [review]
I'm not sure what to make of your comment `For some reason, I was noticing a trailing value in the events generated.`. Did you fix this? Or is this still the case?

And where were you seeing the event? On the extension side or on the event fired by selectBox?

File: src/sql/base/browser/ui/selectBox/selectBox.ts [hunk 3]
```diff
@@ -197,6 +196,10 @@ export class SelectBox extends vsSelectBox {
 	}
 
 	public select(index: number): void {
+		let selectedOptionIndex = this._optionsDictionary.get(this._selectedOption);
+		if (selectedOptionIndex === undefined && selectedOptionIndex === index) { // Not generating an event if the same value is selected.
```
Comment (alanrenmsft) [inline src/sql/base/browser/ui/selectBox/selectBox.ts:200]: === undefined and ===index? you meant to use OR, right?

File: src/sql/base/browser/ui/selectBox/selectBox.ts [hunk 4]
```diff
@@ -134,8 +134,7 @@ export class SelectBox extends vsSelectBox {
 	}
 
 	public onSelect(newInput: ISelectData) {
-		const selected = this._dialogOptions[newInput.index];
-		this._selectedOption = selected.value;
+		this.select(newInput.index);
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/selectBox/selectBox.ts:137]: Pretty sure we don't want to do this - onSelect is called when the base onDidSelect is fired which we already hook in to to fire our own onDidSelect. Calling this.select here will result in two onDidSelect being fired. 

Why did you make this change anyways?

**Review by alanrenmsft — APPROVED** [review]

**Pull request merged**